### PR TITLE
[URP 2D] 2D staging to master

### PR DESCRIPTION
**Pull request by rainsing**
### Purpose of this PR
This PR is an aggregation of several PRs affecting the 2D Renderer:
#1798 #2435 #1990 #2230 #2520 #2495 #1707 #2487 #2524 #1808 #1721 #2568 #2599 #2593 #2630 #2632

---
### Testing status
Each PR is individually tested manually by QA.

---
### Comments to reviewers
Documentation changes will land as a separate PR.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs [hunk 1]
```diff
@@ -279,6 +279,9 @@ internal static class ShaderPropertyId
         public static readonly int sourceTex = Shader.PropertyToID("_SourceTex");
         public static readonly int scaleBias = Shader.PropertyToID("_ScaleBias");
         public static readonly int scaleBiasRt = Shader.PropertyToID("_ScaleBiasRt");
+
+        // Required for 2D Unlit Shadergraph master node as it doesn't currently support hidden properties.
+        public static readonly int rendererColor = Shader.PropertyToID("_RendererColor");
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs:284]: Minor: Maybe this name is too generic for a global constant, might cause conflicts with custom shader properties from users?

**Review by phi-lira — APPROVED** [review]
LGTM. Minor comment about shader property name.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs (hunk 1 above)
Comment (rainsing) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs:284]: This name is used by the engine so we can't really change it here.

**Review by unity-cchu — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]
LGTM on UniversalRP changes. I didn't look in detail at 2D Renderer files though.

**Comment by ellioman** [comment]
Only HDRP fails on ABV which is unrelated to this PR.

**Pull request merged**